src/utils/get-server-status.ts

```diff
@@ -0,0 +1,180 @@
+import { Config } from './../interfaces/config'
+import { sendAlerts } from './alert'
+import { ServerStatus, StatusCount } from './../interfaces/server-status'
+import { Probe } from '../interfaces/probe'
+import { ValidateResponseStatus } from './alert'
+import console, { log } from 'console'
+
+const SERVER_STATUSES: ServerStatus[] = []
+
+// Function to calculate successes and failures, either total or consecutives.
+// If consecutive success/failure equals to threshold, it will send notifications
+const calculateSuccessFailure = async ({
+  errorName,
+  probeStatus,
+  validation,
+  config,
+  request,
+  threshold,
+}: {
+  errorName: string
+  probeStatus: StatusCount
+  validation: ValidateResponseStatus
+  config: Config
+  request: Probe['request']
+  threshold: number
+}) => {
+  log(`Checking for ${errorName} in probe...`)
+  if (probeStatus.isDown === false) {
+    // If current probe is UP and trigger equals true
+    if (validation.status) {
+      // Add failure count
+      probeStatus.consecutiveSuccesses = 0
+      probeStatus.totalFailures += 1
+      probeStatus.consecutiveFailures += 1
+
+      // If count is larger or equal than threshold...
+      // Set status to down, and reset consecutive failure
+      if (probeStatus.consecutiveFailures === threshold) {
+        log(`Sending error notification for "${errorName}"\n`)
+        await sendAlerts({
+          validation,
+          notifications: config.notifications,
+          url: request.url ?? '',
+          status: 'DOWN',
+        })
+        probeStatus.isDown = true
+      }
+    } else {
+      // Server is doing fine
+      probeStatus.consecutiveFailures = 0
+      probeStatus.consecutiveSuccesses += 1
+      probeStatus.totalSuccesses += 1
+    }
+  } else if (probeStatus.isDown === true) {
+    // If current probe is DOWN and trigger equals true
+    if (validation.status) {
+      // Server is not doing fine
+      probeStatus.consecutiveSuccesses = 0
+      probeStatus.consecutiveFailures += 1
+      probeStatus.totalFailures += 1
+    } else {
+      // Add success count
+      probeStatus.consecutiveFailures = 0
+      probeStatus.totalSuccesses += 1
+      probeStatus.consecutiveSuccesses += 1
+
+      // If count is larger or equal than threshold...
+      // Set status to down, and reset consecutive failure
+      if (probeStatus.consecutiveSuccesses === threshold) {
+        log(`Sending resolved notification for "${errorName}"\n`)
+        await sendAlerts({
```

Comment:
The function name is "calculate", why do we have side effect such as sendAlerts in here?

Reply:
Moved out the sendAlerts to looper